R/resultAnnotations.R

```diff
@@ -0,0 +1,925 @@
+#'     annotated UTR regions. Splice types can be one of: 
+#'     annotatedIntron_increasedUsage, annotatedIntron_reducedUsage, 
+#'     exonTruncation, exonElongation, exonTruncation&Elongation, 
+#'     singleExonSkipping, exonSkipping, splicingBeyondGene, 
+#'     multigenicSplicing, downstreamOfNearestGene, upstreamOfNearestGene, 
+#'     complex (everything else).     
+#'     Splice sites (theta metric) annotations indicate how the splice site is 
+#'     located with respect to the reference annotation. The annotated types 
+#'     are: annotatedSpliceSite, exonicRegion, intronicRegion.
+#' 
+#' \code{\link{flagBlacklistRegions}} flags introns in the results table on 
+#'     whether or not they are located in a blacklist region of the genome. By
+#'     default, the blacklist regions as reported in \cite{...} are used.
+#'
+#' @param fds A FraserDataSet
+#' @param txdb A txdb object providing the reference annotation.
+#' @param result A result table as generated by FRASER, including the column 
+#'     \code{annotatedJunction} as generated by the function
+#'     \code{annotateIntronReferenceOverlap}.
+#' @param addSpliceType Logical, indicating if the type of the splice event 
+#'     should be added to the results table. Defaults to \code{TRUE}.
+#' @param addUTRoverlap Logical, indicating if the overlap with UTR regions
+#'     should checked and added to the results table. Defaults to \code{TRUE}.
+#' @param blacklist_regions A BED file that contains the blacklist regions. 
+#'     If \code{NULL} (default), the BED files that are packaged with FRASER 
+#'     are used. 
+#' @param assemblyVersion Indicates the genome assembly version of the intron 
+#'     coordinates. Only used if blacklist_regions is NULL. For other versions, 
+#'     please provide the BED file containing the blacklist regions directly.
+#' @param BPPARAM For controlling parallelization behavior. Defaults to 
+#'     \code{bpparam()}.
+#' @return An annotated FraserDataSet or results table, respectively
+#' 
+#' @name spliceTypeAnnotations
+#' @rdname spliceTypeAnnotations
+#' 
+#' @examples
+#'   # get data, fit and compute p-values and z-scores
+#'   fds <- createTestFraserDataSet()
+#'   
+#'   # load reference annotation
+#'   requireNamespace("TxDb.Hsapiens.UCSC.hg19.knownGene")
+#'   txdb <- 
+#'     TxDb.Hsapiens.UCSC.hg19.knownGene::TxDb.Hsapiens.UCSC.hg19.knownGene 
+#'     
+#'   # add basic annotations for overlap with the reference annotation
+#'   # run this function before creating the results table
+#'   fds <- annotateIntronReferenceOverlap(fds, txdb) 
+#' 
+#'   # extract results: for this small example dataset, only a z score cutoff 
+#'   # of 1 is used to get at least one result. 
+#'   # Make sure to include the additional column in the results table
+#'   res <- results(fds, padjCutoff=NA, zScoreCutoff=1, deltaPsiCutoff=NA, 
+#'                   additionalColumns = 'annotatedJunction')
+#'   res_dt <- as.data.table(res)
```

Comment:
I think we do not have to force our users to use data.table. So if we do not like the output of `results()` or we allow the user to use the results object as is. This means having a check in the beginning of the function to cast the input if not a data.table and back in the end of the function.